client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -329,25 +368,23 @@ public void close() {
         if (closed.getAndSet(true)) {
             return;
         }
-        synchronized (lock) {
+        lock.readLock().lock();
+        try {
             boolean success = false;
-            String sealedSegment = null;
             while (!success) {
                 success = true;
                 for (SegmentOutputStream writer : selector.getWriters()) {
                     try {
                         writer.close();
                     } catch (SegmentSealedException e) {
-                        log.info("Segment was sealed during close: {}", writer);
+                        //Segment sealed exception observed during a close. Re-run close on all the available writers.
```

Comment:
writer.close() invokes writer.flush(). Writer.flush() is waiting for empty inflight events and when it receives a segment sealed exception it implies that the segment sealed call back has executed and updated the writer list.
Hence a wait/sleep was not added.